feedback from stakeholders

danielfmiranda opened the pull request:
Link to review page: https://foundation-s-donate-hel-pq63qo.mofostaging.net/en/donate/help/ (CMS login creds: admin2/admin2)

This PR implements some stakeholder feedback regarding the donate landing,help, and ways-to-give page.

Requests included:
- Being able to create a "ways-to-give" page (OpportunityPage type) underneath the DonateLandingPage ([link to test "ways-to-give" page](https://foundation-s-donate-hel-pq63qo.mofostaging.net/en/donate/ways-to-give/))
- Updating Donate Page navigation bars to be the same as other pages on the foundation site
- Adding the donate banner to donate pages
- Updating the `<p>` element containing the string "Contact Us" on the donate help page to have an id of #contact-us, that way stakeholders can link to this section. (https://github.com/MozillaFoundation/foundation.mozilla.org/issues/11393)

Please see inline comments for more context!

danielfmiranda commented on network-api/networkapi/donate/pagemodels/landing_page.py:
```diff
@@ -13,7 +13,7 @@ class DonateLandingPage(BaseDonationPage):
     parent_page_types: list = ["wagtailpages.Homepage"]
 
-    subpage_types: list = ["DonateHelpPage"]
+    subpage_types: list = ["DonateHelpPage", "wagtailpages.OpportunityPage"]
```
Adding OpportunityPage( AKA "DefaultPage", the page used to create /ways-to-give/), to DonateLandingPage's child page list. 

That way we will be able to create /donate/ways-to-give.

danielfmiranda commented on network-api/networkapi/donate/tests/test_landing_page.py:
```diff
@@ -36,8 +36,8 @@ def test_subpage_types(self):
-            parent_model=pagemodels.DonateLandingPage,
-            child_models={pagemodels.DonateHelpPage},
+            parent_model=DonateLandingPage,
+            child_models={DonateHelpPage, OpportunityPage},
```
Updating the test to account for new child page type. Also, updated file to just name models instead of using `pagemodels.<name_of_donate_page_model>`. Since we are now introducing pages that are not specifically donate related, I thought this would help make the file more readable.

danielfmiranda commented on network-api/networkapi/templates/donate/pages/base.html:
```diff
@@ -30,7 +30,11 @@
 {% block bodyclass %} donate tw-m-0 tw-font-sans tw-font-normal tw-text-base tw-font-normal tw-text-black tw-bg-white tw-scroll-smooth {% endblock %}
 
-{% block donate_banner %}{% endblock %}
+{% block donate_banner %}
```
Per request of @kristinashu, she found that the EOY donate banner was not appearing on donate pages, adding this to the base template will fix that!

danielfmiranda commented on network-api/networkapi/templates/donate/pages/help_page.html:
```diff
@@ -22,7 +22,7 @@ <h1 class="tw-h1-heading"> {{ page.title }} </h1>
             </p>
             <div class="tw-border-t tw-border-b tw-py-10 tw-mt-28 tw-mb-10 tw-border-gray-20">
-                <p class="tw-h3-heading">{% trans "Contact Us" %}</p>
+                <p id="contact-us" class="tw-h3-heading">{% trans "Contact Us" %}</p>
```
Per request of @Miaatmozilla, we are now adding an ID so we can link to the "Contact Us" Section.

jhonatan-lopes commented on network-api/networkapi/templates/donate/pages/base.html:
```diff
@@ -30,7 +30,11 @@
-{% block donate_banner %}{% endblock %}
+{% block donate_banner %}
+    {% if settings.donate_banner.SiteDonateBanner.active_donate_banner %}
+        {% include "partials/donate_banner.html" with banner=settings.donate_banner.SiteDonateBanner.active_donate_banner.localized %}
```
I'm not sure if we should use `.localized` here...  this is how we're doing it in the base, so I guess it's fine, but maybe we should create a ticket to move away from that...

danielfmiranda replied:
Thanks @jhonatan-lopes thats a good point especially after all the work we have done to remove .localized everywhere else in the code base. I have created https://github.com/MozillaFoundation/foundation.mozilla.org/issues/11404 to capture this work 👍

jhonatan-lopes reviewed (approved):
Hey @danielfmiranda, LGTM :+1: I just have one comment but it's not blocking

danielfmiranda commented on network-api/networkapi/templates/donate/pages/base.html:
```diff
@@ -47,5 +51,6 @@
 
 {% block script_bundle %}
     <script src="{% url "javascript-catalog" %}"></script>
+    <script src="{% static "_js/main.compiled.js" %}"></script>
```
Needed to import this JS so the donate banner and navigation would work.

danielfmiranda commented on network-api/networkapi/donate/pagemodels/base_page.py:
```diff
@@ -1,8 +1,8 @@
 
-
-class BaseDonationPage(FoundationMetadataPageMixin, Page):
+class BaseDonationPage(FoundationMetadataPageMixin,FoundationNavigationPageMixin, Page):
```
Needed to add this mixin so the navigation bar will accurately show the links at the top of the page. EG: “[Who we are](https://foundation.mozilla.org/en/who-we-are/), [What we do](https://foundation.mozilla.org/en/what-we-do/), [What we fund](https://foundation.mozilla.org/en/what-we-fund/), [What you can do](https://foundation.mozilla.org/en/what-you-can-do/), [Blog](https://foundation.mozilla.org/en/blog/)”?

danielfmiranda commented on network-api/networkapi/templates/donate/pages/base.html:
```diff
@@ -30,12 +30,16 @@
 
 {% block primary_nav %}
-    {% include "../fragments/nav.html" %}
+    {% include "partials/primary_nav.html" %}
```
Per request of @kristinashu, updating the nav to match the rest of the foundation site! [link to slack thread](https://mozilla.slack.com/archives/C04DYP6T6VD/p1699560239000609)

danielfmiranda commented on source/sass/donate/donate-main.scss:
```diff
@@ -7,6 +7,10 @@
 
 // Custom variables
 @import "../variables";
+@import "../buyers-guide/colors";
+
+// Non-React Components
+@import "../components/primary-nav";
```
SCSS files needed for the nav to render as expected

kristinashu commented:
Looking good! Just a couple things:

The Captcha isn't showing up for me, is that just because it isn't on the real site yet? 

<img width="1157" alt="Screenshot 2023-11-10 at 8 58 50 AM" src="https://github.com/MozillaFoundation/foundation.mozilla.org/assets/1068783/f08f5a9d-fd3c-4b79-87e8-bfb28e5d9ee4">


Not a blocker but ideally the vertical space between between the nav and the heading would be larger so it's the same as on our other pages e.g. screen cap of PR vs Default page

<img width="865" alt="Screenshot 2023-11-10 at 9 00 49 AM" src="https://github.com/MozillaFoundation/foundation.mozilla.org/assets/1068783/15b49ee6-969d-433c-9476-dfdaf22d8a22">

mmmavis reviewed (approved):
LGTM!

mmmavis commented:
@kristinashu the missing recaptcha should be something related to our review app setup. I can't locate where this app is on our Heroku dashboard so I'm unable to confirm at the moment. @danielfmiranda could you please look into this? I can't tell which review app this PR links to as currently Heroku tells me there's none. Thanks!
<img width="408" alt="image" src="https://github.com/MozillaFoundation/foundation.mozilla.org/assets/2896608/dfc665d3-b9e3-4bd4-90f8-1b0a5b6638e9">

kristinashu commented:
Thank you Mavis! Captcha appears on staging https://foundation.mofostaging.net/en/donate/donate-help/ so I'm guessing it will be fine.

<img width="1152" alt="image" src="https://github.com/MozillaFoundation/foundation.mozilla.org/assets/1068783/267c7c85-2323-4890-a10f-3663e7cbe8bc">

danielfmiranda commented:
Thanks @kristinashu and @mmmavis for the review!

To answer the question regarding the captcha, it was due to the CSP settings for review apps. I have updated the settings now and the captcha should appear on new apps created from now on.

Also thank you for the feedback regarding the spacing @kristinashu, I have updated the template to match the spacing from other pages! 

![Screenshot 2023-11-13 at 14-10-38 Donate Help](https://github.com/MozillaFoundation/foundation.mozilla.org/assets/18314510/0d1e0579-1bde-472c-992c-3b496188e075)
